bin/ChakraCore/CMakeLists.txt

```diff
@@ -32,42 +28,82 @@ target_include_directories (
 #  dl: For shared library loading related functions
 #  icuuc: For the ICU (xplat-todo: Make this optional)
 #
-target_link_libraries(ChakraCore
-  -Wl,--no-undefined
-  -Wl,--start-group
-  -Wl,--whole-archive
-  Chakra.Jsrt
-  Chakra.Jsrt.Core
-  -Wl,--no-whole-archive
-  Chakra.Runtime.Types
-  Chakra.Runtime.Math
-  Chakra.Runtime.Library
-  Chakra.Runtime.Language
-  Chakra.Runtime.Debug
-  Chakra.Runtime.ByteCode
-  Chakra.Runtime.PlatformAgnostic
-  Chakra.Runtime.Base
-  Chakra.Parser
-  Chakra.Common.Util
-  Chakra.Common.Memory
-  Chakra.Common.Common
-  Chakra.Common.Core
-  Chakra.Common.DataStructures
-  Chakra.Common.Exceptions
-  Chakra.Common.Codex
-  -Wl,--end-group
-  Chakra.Pal
-  pthread
-  stdc++
-  dl
-  icuuc
-  -Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/libChakraCoreLib.version
-)
-
-#
-# Post build step to copy the built shared library
-# to BuildLinux (or whatever the CMakeBuildDir is)
-add_custom_command(TARGET ChakraCore POST_BUILD
-  COMMAND ${CMAKE_COMMAND} -E copy_if_different 
-  "${CHAKRACORE_BINARY_DIR}/bin/ChakraCore/libChakraCore.so"
-   ${CHAKRACORE_BINARY_DIR}/)
+if(CMAKE_SYSTEM_NAME STREQUAL Linux)
+    target_link_libraries(ChakraCore
+      -Wl,--no-undefined
+      -Wl,--start-group
+      -Wl,--whole-archive
+      Chakra.Jsrt
+      Chakra.Jsrt.Core
+      -Wl,--no-whole-archive
+      Chakra.Runtime.Types
+      Chakra.Runtime.Math
+      Chakra.Runtime.Library
+      Chakra.Runtime.Language
+      Chakra.Runtime.Debug
+      Chakra.Runtime.ByteCode
+      Chakra.Runtime.PlatformAgnostic
+      Chakra.Runtime.Base
+      Chakra.Parser
+      Chakra.Common.Util
+      Chakra.Common.Memory
+      Chakra.Common.Common
+      Chakra.Common.Core
+      Chakra.Common.DataStructures
+      Chakra.Common.Exceptions
+      Chakra.Common.Codex
+      -Wl,--end-group
+      Chakra.Pal
+      pthread
+      stdc++
+      dl
+      icuuc
+      -Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/libChakraCoreLib.version
+    )
+    #
+    # Post build step to copy the built shared library
+    # to BuildLinux (or whatever the CMakeBuildDir is)
+    add_custom_command(TARGET ChakraCore POST_BUILD
+      COMMAND ${CMAKE_COMMAND} -E copy_if_different
+      "${CHAKRACORE_BINARY_DIR}/bin/ChakraCore/libChakraCore.so"
+       ${CHAKRACORE_BINARY_DIR}/)
+elseif(CMAKE_SYSTEM_NAME STREQUAL Darwin)
```

Comment:
One thing is that its a little hard to read what the difference is between OSX and Linux- can we take this list, assign it to a CMakeList variable and then append just the diffs in the OSX/Linux cases (and then assign the customized list to target_link_libraries). If not, a comment on what the differences are in this particular case between OSX and Linux would make this easier to read in the future.